### PR TITLE
Add sagelib

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
@ocefpaf ready for review

File: recipes/sagelib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    - readline 6.2*
+    - rw
+    - singular
+    - six
+    - symmetrica
+    - zn_poly
+
+test:
+  import:
+    - sage
```
Comment (isuruf) [inline recipes/sagelib/meta.yaml]: Cannot import sage.all as all the runtime dependencies are not in conda-forge and some runtime dependencies need sagelib to build.

**Comment by isuruf** [comment]
ping

**Comment by isuruf** [comment]
Thanks

**Comment by ocefpaf** [comment]
No problem  ;-)

File: recipes/sagelib/build.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export CXXFLAGS="-w $CXXFLAGS"
+
+export SAGE_FAT_BINARY=yes
+export SAGE_LOCAL=$PREFIX
+ln -s $PREFIX local
+export SAGE_NUM_THREADS=2
+
+# make build builds sagelib and copies some scripts
+# the scripts can be moved to another package and call it sageruntime
+make build
```
Comment (saraedum) [inline recipes/sagelib/build.sh]: Why do you need to call a patched make build? Couldn't you just invoke `setup.py` instead?

Comment (isuruf) [inline recipes/sagelib/build.sh]: Is that possible? I thought there were files generated by the `Makefile`. If they are not needed for sagelib, we can just use `setup.py`

**Pull request merged**